### PR TITLE
Feature 83 jquery html gen and callbacks in genplayerhtml

### PR DESCRIPTION
**Pull request by stheid**
This is a big one. @ApolloLV please test!

I tested it and i think its a big improvement to the client code

be aware, the disabled buttons are on purpose.

changes
 - spectators can no longer be made throwers
 - spectators have no abgabe button
 - players cannot verify abgaben of their team including themselfs
 - players can only rejoin if abgegeben if there are strafbiere in the team
 - players can only rejoin THEMSELFS
 - rejoining automatically reduces strafbiercount

This code is also related to the umlaut topic, but it is probably only partially solved. i would start on that issue once this code is fully tested in the develop!

### CODE REVIEW

File: client/public/js/client.js [hunk 1]
```diff
@@ -294,21 +294,22 @@ function processNewState(state) {
...
+        $('#teambarea').append(generatePlayerHTML(player, currentGameState.throwingplayer, player.team === playerTeam, currentGameState.strafbierteamb));
         if (player.name === currentGameState.throwingplayer) {
             currentTeam = EnumTeams.TEAM_B_TEAMS;
         }
     });
     currentGameState.spectatorsList.forEach(function (player, index) {
         player.team = EnumTeams.SPECTATOR_TEAMS
-        $('#spectatorarea').append(generatePlayerHTML(player, currentGameState.throwingplayer, false));
+        // This is ugly, but js does not support named parameters, therefore we have to fill the other parameters with undefineds
+        $('#spectatorarea').append(generatePlayerHTML(player, undefined, undefined, undefined, isSpectator = true));
```
Comment (ApolloLV) [inline client/public/js/client.js:312]: if js does not have named parameters, why do you call it with isSpectator=true?

File: client/public/js/client.js [hunk 2]
```diff
@@ -506,83 +505,95 @@ function changeLowBandwidthMode() {
...
+    egoClass = isHimself ? ' egoplayer' : '';
+    hasAbgegebenClass = player.abgegeben ? ' disabled' : '';
+
+    // disabled for own team and not abgegeben
+    // disabled if abgegeben but has no strafbier or is not himself
+    disableAbgabeButtonClass = ''
+    if (isOwnTeam && !player.abgegeben
+        || player.abgegeben && (!hasStrafbier || !isHimself)) {
+        disableAbgabeButtonClass = " disabled"
+    }
```
Comment (ApolloLV) [inline client/public/js/client.js:521]:
```suggestion
    if (isOwnTeam && !player.abgegeben
        || player.abgegeben && (!hasStrafbier || !isHimself)) {
        disableAbgabeButtonClass = " disabled";
    }else{
         disableAbgabeButtonClass = '';
     }
```

**Comment by moreApi** [comment]
Do we also want to enforce those restrictions in the server?

**Comment by stheid** [comment]
for now i reverted the restrictions. maybe lets discuss it in #89

**Pull request merged**